File: solutions/csharp/pascals-triangle/PascalsTriangle.cs
using System;
using System.Collections.Generic;

public static class PascalsTriangle
{
    public static IEnumerable<IEnumerable<int>> Calculate(int rows)
    {
        var result = new List<int[]> { };

        while (rows > 0)
        {
            if (result.Count == 0)
            {
                result.Add(new int[] { 1 });
            }
            else
            {
                var nextRow = new List<int> { 1, 1 };
                result.Add(nextRow.ToArray());
            }
            rows -= 1;
        }

        return result.ToArray();
    }

}

File: solutions/dart/pascals-triangle/lib/pascals_triangle.dart
 class PascalsTriangle {
  List<List<int>> rows(int n) {
    List<List<int>> triange = [];
    List<int> row = [1];

    while (n > 0) {
      triange.add(row);
      row = _generateNextRow(row);
      n--;
    }
    return triange;
  }

  List<int> _generateNextRow(List<int> previous) {
    var nextRow = [1];
    for (var i = 1; i < previous.length; i++) {
      nextRow.add(previous[i-1] + previous[i]);
    }
    nextRow.add(1);
    return nextRow;
  }
}
